test/infrastructure/docker/controllers/dockermachine_controller.go

```diff
@@ -197,6 +197,10 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, machine *
 	// Make sure bootstrap data is available and populated.
 	if machine.Spec.Bootstrap.DataSecretName == nil {
 		log.Info("Waiting for the Bootstrap provider controller to set bootstrap data")
+		if !cluster.Status.ControlPlaneInitialized {
```

Comment:
`WaitingForControlPlaneAvailableReason` provides better information about why we are reading for bootstrap data on a machine